src/main/java/com/google/sps/data/LoggedUser.java

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+/** Class containing a logged user. */
+public final class LoggedUser extends User {
+
+  private final String id;
+  private Long usedSpace;
+  private final String logoutURL;
+
+  public LoggedUser(String id, String logoutURL) {
+    // Init User super class with loggedIn = true.
+    super(true);
+
+    this.id = id;
+    this.logoutURL = logoutURL;
+
+    // Get user's usedSpace from the database.
+    this.usedSpace = (long) getEntity().getProperty("usedSpace");
+  }
+
+  /** Returns the datastore BlurImage entities corresponding to the user ordered by dateCreated. */
+  public Iterable<Entity> getBlurImageEntities(int maxResults) {
+    if (maxResults < 0) {
+      maxResults = 0;
+    }
+
+    // Load user's photos from datastore.
+    Query query = new Query("BlurImage");
+    query.setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, id));
+    query.addSort("dateCreated", SortDirection.DESCENDING);
```

Comment:
Curious if there is a way to get 'maxResults' in the query itself.

Reply:
The way I made it is pretty standard and AFAIK there is no way to get it in the query itself.